### ISSUE TITLE
False positives in reportOptionalOperand and reportOperatorIssue

### ISSUE THREAD

**Issue by rdegraaf** [microsoft/pyright#10046]
Pylance generates the noted warnings on the code below:

```
def foo(s: typing.Optional[int]) -> int:
    # Operator "<" not supported for types "Literal[0]" and "int | None" Pylance(reportOperatorIssue) v
    #   Operator "<" not supported for types "Literal[0]" and None"
    assert None is s or 0 < s
    if None is s:
        return False
    else:
        # Operator "+" not supported for "None" Pylance(reportOptionalOperand)
        return s + 1 
```
As far as I can tell, all of the warnings are false positives as the target code can only be reached if `s` is not `None`.  I get the same behaviour if I replace "`typing.Optional[int]`" with "`typing.Union[int, None]`".  If I flip the "`0 < s`" comparison around to "`s > 0`", then I instead get the warning "`Operator ">" not supported for "None" Pylance(reportOptionalOperand)`"

I'm using Pylance 2025.3.1 within Visual Studio Code 1.97.2.
